### ISSUE TITLE
[java] MissingStaticMethodInNonInstantiatableClass false positive with method inner classes

### ISSUE THREAD

**Issue by hgschmie** [pmd/pmd#3701]
**Affects PMD Version: 6.41.0** 

**Rule: MissingStaticMethodInNonInstantiatableClass**

Please provide the rule name and a link to the rule documentation:
https://pmd.github.io/latest/pmd_rules_java_errorprone.html#missingstaticmethodinnoninstantiatableclass

**Description:**

Using a method nested class causes a false positive: 

PMD Failure: Scratch$InnerClass:11 Rule:MissingStaticMethodInNonInstantiatableClass Priority:3 Class cannot be instantiated and does not provide any static methods or fields.

**Code Sample demonstrating the issue:**

```java
public class Scratch {

    public static void main(String[] args) {
        Scratch scratch = new Scratch();

        scratch.callMethod();
    }

    void callMethod() {

        class InnerClass {
            private InnerClass() {
            }

            void display() {
                System.out.println("Works OK!");
            }
        }

        InnerClass innerClass = new InnerClass();
        innerClass.display();
    }
}
```

This code compiles just fine and running it prints `Works OK!`.

**Expected outcome:**

PMD understands this code just fine

PMD reports a violation at line 11, but that's wrong. That's a false positive.

**Running PMD through:** *[Maven]*
